test/ng/directive/ngModelSpec.js

```diff
@@ -1221,6 +1221,80 @@ describe('ngModel', function() {
         expect(ctrl.$validators.mock).toHaveBeenCalledWith('a', 'ab');
         expect(ctrl.$validators.mock.calls.length).toEqual(2);
       });
+
+      it('should validate correctly when $parser name equals $validator key', function() {
+
+        var testValid, otherValid, parseValid;
+
+        ctrl.$validators.test = function(value) {
+          return testValid;
+        };
+
+        ctrl.$validators.other = function(value) {
+          return otherValid;
+        };
+
+        ctrl.$$parserName = 'test';
```

Comment:
Any ideas? It's currently a private API, but we rely on it for the name of the error key. A separate function, $setParseError(name) might be possible that is called inside the parse error. Although I think we have to stick with `return undefined` denoting a parse error, bad as it is.